docs/admin/authentication.md

```diff
@@ -222,44 +222,103 @@ from the OAuth2 [token response](https://openid.net/specs/openid-connect-core-1_
-configured to trust a single client. This allows the use of public providers,
+configured to trust a single issuer. This allows the use of public providers,
 such as Google, without trusting credentials issued to third parties. Admins who
 wish utilize multiple OAuth clients should explore providers which support the
 `azp` (authorized party) claim, a mechanism for allowing one client to issue
 tokens on behalf of another.
 
-The plugin also accepts the following optional flags:
-
-* `--oidc-ca-file` Used by the API server to establish and verify the secure
-connection to the issuer. Defaults to the host's root CAs.
-
-And experimental flags:
-
-* `--oidc-username-claim` JWT claim to use as the user name. By default `sub`,
-which is expected to be a unique identifier of the end user. Admins can choose
-other claims, such as `email`, depending on their provider.
-* `--oidc-groups-claim` JWT claim to use as the user's group. If the claim is present
-it must be an array of strings.
-
 Kubernetes does not provide an OpenID Connect Identity Provider.
 You can use an existing public OpenID Connect Identity Provider (such as Google, or [others](http://connect2id.com/products/nimbus-oauth-openid-connect-sdk/openid-connect-providers)).
-Or, you can run your own Identity Provider, such as CoreOS [dex](https://github.com/coreos/dex), [Keycloak](https://github.com/keycloak/keycloak) or CloudFoundry [UAA](https://github.com/cloudfoundry/uaa).
+Or, you can run your own Identity Provider, such as CoreOS [dex](https://github.com/coreos/dex), [Keycloak](https://github.com/keycloak/keycloak), CloudFoundry [UAA](https://github.com/cloudfoundry/uaa), or Tremolo Secutity's [OpenUnison](https://github.com/tremolosecurity/openunison).
+
+For an identity provider to work with Kubernetes it must:
+
+1.  Support [OpenID connect discovery](https://openid.net/specs/openid-connect-discovery-1_0.html); not all do.
+2.  Run in TLS with non-obsolete ciphers
+3.  Have a CA signed certificate (even if the CA is not a commercial CA or is self signed)
 
-The provider needs to support [OpenID connect discovery](https://openid.net/specs/openid-connect-discovery-1_0.html); not all do.
+A note about requirement #3 above, requiring a CA signed certificate.  If you deploy your own identity provider (as apposed to one of the cloud providers like Goolge or Microsoft) you MUST have your identity provider's web server certificate signed by a CA.  This is an issue with GoLang's TLS client implementation not being able to verify a self-signed certificate.  If you don't have a CA handy, you can use this script from the CoreOS team to create a simple CA and a signed certificate and key pair - https://github.com/coreos/dex/blob/1ee5920c54f5926d6468d2607c728b71cfe98092/examples/k8s/gencert.sh or this script based on it that will generate sha256 certs with a longer life and larger keysize https://raw.githubusercontent.com/TremoloSecurity/openunison-qs-kubernetes/master/makecerts.sh.
```

Comment:
> This is an issue with GoLang's TLS client implementation not being able to verify a self-signed certificate.

that's not strictly true... a self-signed cert works as long as the cert is allowed to be used as a CA (CA:TRUE, etc)